pkg/operator/kit/resource.go

```diff
@@ -22,140 +22,162 @@ package kit
-
-	// RetryDelay is the number of seconds to delay between retrying of kubernetes API calls.
-	// Only used by the Retry function.
-	RetryDelay int
-
-	// MaxRetries is the number of times that an operation will be attempted by the Retry function.
-	MaxRetries int
-
-	// The host where the Kubernetes master is found.
-	MasterHost string
-
-	// An http connection to the Kubernetes API
-	KubeHTTPCli *http.Client
-}
-
 // CustomResource is for creating a Kubernetes TPR/CRD
 type CustomResource struct {
 	// Name of the custom resource
 	Name string
 
+	// Plural of the custom resource in plural
+	Plural string
+
 	// Group the custom resource belongs to
 	Group string
 
 	// Version which should be defined in a const above
 	Version string
 
-	// Description that is human readable
-	Description string
+	// Scope of the CRD. Namespaced or cluster
+	Scope apiextensionsv1beta1.ResourceScope
+
+	// Kind is the serialized interface of the resource.
+	Kind string
 }
 
-// CreateCustomResources creates the given custom resources and waits for them to initialize
-func CreateCustomResources(context KubeContext, resources []CustomResource) error {
-	for _, resource := range resources {
-		if err := CreateCustomResource(context, resource); err != nil {
-			return fmt.Errorf("failed to init resource %s. %+v", resource.Name, err)
-		}
+// CreateCustomResource creates a single custom resource,and wait for it to be active
+// The resource is of kind CRD if the Kubernetes server is 1.7.0.
+// The resource is of kind TPR if the Kubernetes server is before 1.7.0.
+func CreateCustomResource(resource CustomResource, clientset kubernetes.Interface) error {
+
+	// CRD is available on v1.7.0. TPR became deprecated on v1.7.0
+	serverVersion, err := clientset.Discovery().ServerVersion()
+	if err != nil {
+		return fmt.Errorf("Error getting server version: %v", err)
 	}
+	kubeVersion := version.MustParseSemantic(serverVersion.GitVersion)
 
-	for _, resource := range resources {
-		if err := waitForCustomResourceInit(context, resource); err != nil {
-			return fmt.Errorf("failed to complete init %s. %+v", resource.Name, err)
+	if kubeVersion.AtLeast(version.MustParseSemantic("v1.7.0")) {
+		return createCRD(resource)
+	}
+
+	// Create TPR for Kubernetes version less than v1.7.0
+	return createTPR(resource, clientset)
+}
+
+func createCRD(resource CustomResource) error {
+	logger.Infof("creating %s CRD", resource.Name)
+	crdName := fmt.Sprintf("%s.%s", resource.Plural, resource.Group)
+	crd := &apiextensionsv1beta1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: crdName,
+		},
+		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
+			Group:   resource.Group,
+			Version: resource.Version,
+			Scope:   resource.Scope,
+			Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+				Singular: resource.Name,
+				Plural:   resource.Plural,
+				Kind:     resource.Kind,
+			},
+		},
+	}
+	clientset, err := getApiextensionsClientset()
```

Comment:
The kit didn't ever load its own context, it always accepted an instance of kubernetes.Interface. Likewise, the kit should require an instance of the new interface, which the caller passes. 
Until we add the operator kit to glide we can just make changes here.